src/main/groovy/edu/oregonstate/mist/cardsapi/resources/CardsResource.groovy

```diff
@@ -65,4 +82,129 @@ class CardsResource extends Resource {
         }
 
     }
+
+    //Get cards by parameters
+    @GET
+    @Path ('')
+    @Produces(MediaType.APPLICATION_JSON)
+    Response getCards(@QueryParam("types") List<String> types,
+                      @QueryParam("name") Optional<String> name,
+                      @QueryParam("colors") List<String> colors,
+                      @QueryParam("rarities") List<String> rarities,
+                      @QueryParam("energyMin") Optional<Integer> energyMin,
+                      @QueryParam("energyMax") Optional<Integer> energyMax,
+                      @QueryParam("keywords") List<String> keywords,
+                      @QueryParam("number") Optional<Integer> number,
+                      @QueryParam("isRandom") Optional<Boolean> isRandom) {
+
+        if(!types) {
+            types = validTypes
+        } else {
+            List<String> invalidTypes = types - validTypes
+            if(invalidTypes) {
+                return badRequest("Invalid types: \'${invalidTypes.join("\', \'")}\'. " +
+                        "Valid types are: " +
+                        "skill, attack, power, status, curse").build()
+            }
+        }
+
+        if(!colors) {
+            colors = validColors
+        } else {
+            List<String> invalidColors = colors - validColors
+            if(invalidColors) {
+                return badRequest("Invalid colors: \'${invalidColors.join("\', \'")}\'. " +
+                        "Valid colors are: " +
+                        "red, green, blue, colorless.").build()
+            }
+        }
+
+        if(!rarities) {
+            rarities = validRarities
+        } else {
+            List<String> invalidRarities = rarities - validRarities
+            if(invalidRarities) {
+                return badRequest("Invalid rarities: \'${invalidRarities.join("\', \'")}\'. " +
+                        "Valid rarities are: " +
+                        "basic, common, uncommon, rare").build()
+            }
+        }
+
+        if(!(name.or("")).matches(regEx)) {
+            return badRequest("Invalid name: \'" + name.get() +
+                    "\'. Name must match pattern: " +
+                    regEx).build()
+        }
+
+        for(int i = 0; i < keywords.size(); i++) {
+            if(!keywords[i].matches(regEx)) {
+                return badRequest("Invalid keyword: \'" + keywords[i] +
+                        "\'. All keywords " +
+                        "must match pattern: " + regEx).build()
+            }
+        }
+
+        List<Card> cards = cardFluent.getCards(types, name.orNull(), colors, rarities,
+                energyMin.or(0), energyMax.or(999),
+                keywords, number.or(10), isRandom.or(true))
+
+        ResultObject cardResult = cardsResult(cards)
+        ok(cardResult).build()
+    }
+
+    @POST
+    @Path('')
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces(MediaType.APPLICATION_JSON)
+    Response postCard (@Valid Card newCard) {
```

Comment:
In your swagger spec, it looks like you are posting a NewData object (like the resultobject) but that's not the case here. We should be accepting a resultobject as the Post body to adhere to jsonapi spec.